velox/docs/stats.rst

```diff
@@ -0,0 +1,126 @@
+==============
+Stats Counters
+==============
+Stats counter is used to collect the metrics of important velox runtime events
+for monitoring purpose. The collected metrics can provide insights into the
+continuous availability and performance analysis of a velox runtime system. For
+instance, the collected data can help automatically generate alerts at an
+outage. Velox provide a framework to collect the customized stats which consists
+of three steps:
+
+**Stats Definition**: define the name, type and config parameters for each stats,
+and register through registerVeloxCounters() on velox runtime startup.
+
+**Stats Record**: when the event of a predefined stats happens in production, the
+velox runtime records the stats data point through the stats collection
+interface provided by BaseStatsReporter.
+
+**Stats Export**: BaseStatsReporter aggregates the collected data points based
+on the registered stats definition, and periodically exports to the backend
+monitoring service. The implementation of BaseStatsReporter is based on the
+actually integrated monitoring service, such as ODS used by Meta, OpenCensus and
+Prometheus provided by OSS. The stats aggregation granularity and export
+interval are also configured based on the actual used monitoring service.
+
+Velox provides the following five different types of stats defined by StatType
+enum class:
+
+**Count**: tracks the count of events, such as the number of query failures.
+Count stats type is registered through REPORT_ADD_STAT_EXPORT_TYPE and recorded
+by REPORT_ADD_STAT_VALUE.
+
+**Sum**: tracks the sum of event data point values, such as sum of query scan
+read bytes. Sum stats type is registered through REPORT_ADD_STAT_EXPORT_TYPE
```

Comment:
The names of REPORT_ADD_STAT_EXPORT_TYPE and REPORT_ADD_STAT_VALUE macros are a bit strange. Wouldn't these be clearer as DEFINE_COUNTER and RECORD_COUNTER to match the actions above?